easybuild/easyblocks/u/ucx_plugins.py

```diff
@@ -43,16 +44,37 @@ class EB_UCX_Plugins(ConfigureMake):
     """Support for building additional plugins for a existing UCX module"""
 
     def __init__(self, *args, **kwargs):
-        """Custom initialization for UCX: set correct module name."""
+        """Custom initialization for UCX-Plugins."""
         super(EB_UCX_Plugins, self).__init__(*args, **kwargs)
-        self.plugins = {}
+        self._plugins = None
         self.makefile_dirs = []
 
+    @property
+    def plugins(self):
+        """Property to determine list of plugins based on loaded dependencies, or return cached list of plugins."""
+        if self._plugins is None:
+            plugins = defaultdict(list)
+            dep_names = set(dep['name'] for dep in self.cfg.dependencies())
```

Comment:
@Flamefire Let's update this to use `self.cfg.dependency_names()`, now that https://github.com/easybuilders/easybuild-framework/pull/4360 is merged?

Reply:
```suggestion
            dep_names = cfg.dependency_names()